mubench.pipeline/tasks/implementations/dataset_check.py

```diff
@@ -0,0 +1,213 @@
+            if "file" not in location:
+                self._missing_key("location.file", yaml_path)
+            if "method" not in location:
+                self._missing_key("location.method", yaml_path)
+
+        if "api" not in misuse_yaml:
+            self._missing_key("api", yaml_path)
+
+        if not misuse_yaml.get("characteristics", None):
+            self._missing_key("characteristics", yaml_path)
+
+        if not misuse_yaml.get("description", None):
+            self._missing_key("description", yaml_path)
+
+        if "fix" not in misuse_yaml:
+            self._missing_key("fix", yaml_path)
+        else:
+            fix = misuse_yaml["fix"]
+            if "commit" not in fix:
+                self._missing_key("fix.commit", yaml_path)
+            if "revision" not in fix:
+                self._missing_key("fix.revision", yaml_path)
+
+        if "internal" not in misuse_yaml:
+            self._missing_key("internal", yaml_path)
+
+        if "report" not in misuse_yaml:
+            self._missing_key("report", yaml_path)
+
+        if "source" not in misuse_yaml:
+            self._missing_key("source", yaml_path)
+        else:
+            source = misuse_yaml["source"]
+            if "name" not in source:
+                self._missing_key("source.name", yaml_path)
+            if "url" not in source:
+                self._missing_key("source.url", yaml_path)
+
+    def _check_misuses_listed_in_version_exist(self, version: ProjectVersion):
+        for misuse_id in version._yaml.get("misuses", []) or []:
+            if misuse_id not in [misuse.misuse_id for misuse in version.misuses]:
+                self._unknown_misuse(version.id, misuse_id)
+
+    def _check_version_misuse_id_conflict(self, project_id: str, version_id: str, misuse_id: str):
+        if version_id == misuse_id:
+            self._version_misuse_conflict(project_id, version_id)
+
+    def _check_misuse_location_exists(self, project: Project, version: ProjectVersion, misuse: Misuse):
+        if "location" in misuse._yaml:
+            location = misuse.location
+            if location.file and location.method:
+
+                checkout = version.get_checkout(self.checkout_base_path)
+                if not checkout.exists():
+                    self.logger.info(
+                            'Cannot check location for "{}": requires checkout of "{}".'.format(
+                                misuse.id, version.id))
+                    return
+
+                source_base_path = join(checkout.checkout_dir, version.source_dir)
+                if not self._location_exists(source_base_path, location.file, location.method):
+                    self._cannot_find_location(str(location), "{}/misuses/{}/misuse.yml".format(project.id, misuse.misuse_id))
+
+    def _location_exists(self, source_base_path, file_, method) -> bool:
+        return get_snippets(source_base_path, file_, method)
+
+    def _new_known_datasets_entry(self, id_: str):
```

Comment:
`register` or `add` instead of `new`? `legal` instead of `known`? Not 100% happy with my suggestions, but also not with the current name.